### PR TITLE
NAS-116405 / 22.12 / Added details card widget

### PR DESCRIPTION
**Pull request by RehanY147**
Check the details card widget under Datasets page.
Check the new loading card info component. it should show a blinking/fade-in-fade-out animation
Check that the details card should only show up for dataset rows and not for pool rows
Also test the `Edit` and `Delete` buttons

### CODE REVIEW

**Review by undsoft — APPROVED** [review]
Nice, but I think we can go even more granular in some cases.
For example here: https://jira.ixsystems.com/browse/NAS-116416 most of the content is static with only `Password is set` being a dynamic string. We could show loading indication for those strings specifically.
Also something to think about is error handling. If request for Global SED Password fails, it may be ? a good idea to show error inline with a retry button.

Approving because what you have here may be useful for some cards.

**Comment by dszidi** [comment]
So far looks pretty good. 

 > Check that the details card should only show up for dataset rows and not for pool rows

 Why only show details for dataset rows? The first row is the root dataset and should also be showing details

File: src/app/pages/datasets/components/dataset-details-panel/dataset-details-panel.component.html [hunk 1]
```diff
@@ -1,7 +1,8 @@
-<h3>Details for {{ dataset.name }}</h3>
+<h3 class="title">Details for {{ dataset.name }}</h3>
 
 <div class="cards">
   <ix-permissions-card [dataset]="dataset"></ix-permissions-card>
+  <ix-dataset-details-card *ngIf="dataset.id !== dataset.pool" [dataset]="dataset"></ix-dataset-details-card>
```
Comment (dszidi) [inline src/app/pages/datasets/components/dataset-details-panel/dataset-details-panel.component.html:5]: This condition should be removed. The root dataset has the same name as the pool and has details we need to show

**Comment by RehanY147** [comment]
Showing `1TiB of 3TiB` space data doesn't work for ZFS according to middleware. We can show used or available space but not total space. @dszidi @undsoft Should we remove that attr completely or change it to show one of the two options possible?

**Comment by anodos325** [comment]
> Showing `1TiB of 3TiB` space data doesn't work for ZFS according to middleware. We can show used or available space but not total space. @dszidi @undsoft Should we remove that attr completely or change it to show one of the two options possible?

As mentioned in the ticket, I believe that the Dataset Details should include breakout of used stats. Example:
```
used: 776K (dataset: 144K, snapshots: 320K, children: 312K)
```

We'll update the `pool.dataset.query` return to include this info.

**Comment by dszidi** [comment]
@RehanY147 @undsoft Unfortunately this branch is not compiling for me so I can't really comment on the animation issues.

Wrt the total storage stat. We can possibly break things down as @anodos325 suggested but put it in the capacity management card instead of the dataset details card. The layout is already busting at the seams, so the only way I can see that working is by ditching the little chart and going text only.

As for the overview section, perhaps we can just go with the used stat since that is what we highlight in the storage dashboard usage card. Users are probably most concerned with hitting that 80% usage mark. Perhaps it would be wise to prioritize that in the datasets overview as well. This would also cut down on screen real estate

@undsoft @RehanY147 @anodos325 what do you think? ^^

**Comment by anodos325** [comment]
> @RehanY147 @undsoft Unfortunately this branch is not compiling for me so I can't really comment on the animation issues.
> 
> Wrt the total storage stat. We can possibly break things down as @anodos325 suggested but put it in the capacity management card instead of the dataset details card. The layout is already busting at the seams, so the only way I can see that working is by ditching the little chart and going text only.
> 
> As for the overview section, perhaps we can just go with the used stat since that is what we highlight in the storage dashboard usage card. Users are probably most concerned with hitting that 80% usage mark. Perhaps it would be wise to prioritize that in the datasets overview as well. This would also cut down on screen real estate
> 
> @undsoft @RehanY147 @anodos325 what do you think? ^^

I know from forums interactions there have been many cases where users are quite confused about why their USED is really high even after deleting a bunch of files. Showing how it's used may be incredibly helpful for them and also avoid some Tier 1 support calls.

**Comment by RehanY147** [comment]
I will remove remnants of custom info loading components if we like the `ngx-skeleton-loader`.

**Review by undsoft — APPROVED** [review]
I think we should keep the 3rd party package and adjust animation.
For example, setting opacity: 0.25 makes it better for me on dark theme.
We should also consider extracting as much as possible to `forRoot`. `findTheme` can become `var(--alt-bg2)`

File: src/app/pages/datasets/components/dataset-details-card/dataset-details-card.component.html [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+          <span *ngIf="!loading || !!datasetCompression">{{ datasetCompression }}</span>
+          <ngx-skeleton-loader
+            *ngIf="loading && !datasetCompression"
+            class="loader-skel" 
+          ></ngx-skeleton-loader>
+        </div>
+        <div class="row">
+          <span class="header">Enable Atime: </span>
+          <span *ngIf="!loading || !!dataset.atime?.value">{{ dataset.atime?.value }}</span>
+          <ngx-skeleton-loader
```
Comment (denysbutenko) [inline src/app/pages/datasets/components/dataset-details-card/dataset-details-card.component.html:35]: It makes sense to wrap `ngx-skeleton-loader` in a component with common styles and etc

File: src/app/pages/datasets/components/dataset-details-card/dataset-details-card.component.html [hunk 3]
```diff
@@ -0,0 +1,66 @@
+<div>
+    <mat-card class="container-card">
+      <mat-card-header>
+        <h3 mat-card-title>{{ 'Dataset Details' | translate }}</h3>
+        <button mat-button (click)="editDataset()">{{ 'Edit' | translate }}</button>
+      </mat-card-header>
+      <mat-card-content>
+        <div class="row">
+          <span class="header">Type: </span>
```
Comment (denysbutenko) [inline src/app/pages/datasets/components/dataset-details-card/dataset-details-card.component.html:9]: Missing translate pipe on labels

File: src/app/pages/datasets/components/dataset-details-card/dataset-details-card.component.html (hunk 2 above)
Comment (undsoft) [reply src/app/pages/datasets/components/dataset-details-card/dataset-details-card.component.html:35]: Let's fix this later. Extracting to `forRoot` should be good enough for now.

**Pull request merged**